x-pack/plugins/data_visualizer/public/application/index_data_visualizer/index_data_visualizer.tsx

```diff
@@ -41,16 +43,44 @@ export interface DataVisualizerUrlStateContextProviderProps {
   additionalLinks: ResultLink[];
 }
 
+export const getLocatorParams = (params: {
+  indexPatternId?: string;
+  savedSearchId?: string;
+  searchString: string;
+  searchSessionId?: string;
+  shouldRestoreSearchSession: boolean;
+}): IndexDataVisualizerLocatorParams => {
+  const urlState = parseUrlState(params.searchString);
+
+  let locatorParams: IndexDataVisualizerLocatorParams = {
+    indexPatternId: urlState.index,
+    searchSessionId: params.searchSessionId,
+  };
+
+  if (urlState) {
```

Comment:
I am having issues restoring a search session if the session was created off a data visualizer view that was opened off a saved search (rather than a data view). The data visualizer page opens up empty, with no errors on the page or in the console.

![image](https://user-images.githubusercontent.com/7405507/146762630-3a684451-fc82-4b2e-a0f1-1a604ce929a2.png)

Reply:
Fixed here https://github.com/elastic/kibana/pull/121053/commits/a2359f1bc1e9630f67965f07fc3b7f667dfc0ba4